cmd/gdeploy/main.go

```diff
@@ -268,3 +270,36 @@ func PreventDevUsage() cli.BeforeFunc {
 		return nil
 	}
 }
+
+// Check whether gdeploy version matches with the granted-approval.
+func VerifyGDeployCompatibility() cli.BeforeFunc {
+	return func(c *cli.Context) error {
+		ctx := c.Context
+		dc, err := deploy.ConfigFromContext(ctx)
+		if err != nil {
+			return err
+		}
+
+		// skip compatibility check for dev deployments.
+		if dc.Deployment.Dev != nil && *dc.Deployment.Dev {
+			return nil
+		}
+
+		isValidReleaseNumber, err := regexp.MatchString(`v\d.\d+.\d+`, dc.Deployment.Release)
+		if err != nil {
+			return err
```

Comment:
I think we can omit this check entirely, we only need to know if the release number matches the build.Version and if not inform teh user with an error message showing what they had vs what was expected